src/Sentry/Internal/TcpPing.cs

```diff
@@ -2,21 +2,29 @@ namespace Sentry.Internal;
 
 internal interface IPing
 {
-    Task<bool> IsAvailableAsync();
+    Task<bool> IsAvailableAsync(CancellationToken cancellationToken);
 }
 
 internal class TcpPing(string hostToCheck, int portToCheck = 443) : IPing
 {
     private readonly Ping _ping = new();
 
-    public async Task<bool> IsAvailableAsync()
+    public async Task<bool> IsAvailableAsync(CancellationToken cancellationToken)
     {
         try
         {
             using var tcpClient = new TcpClient();
+#if NET5_0_OR_GREATER
+            await tcpClient.ConnectAsync(hostToCheck, portToCheck, cancellationToken).ConfigureAwait(false);
+#else
             await tcpClient.ConnectAsync(hostToCheck, portToCheck).ConfigureAwait(false);
+#endif
```

Comment:
That means sub .NET5 there's no cancellation?

Reply:
tcpClient.ConnectAsync doesn't allow passing a cancellation token in netstandard or netfx no.